src/components/BookList.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import books from './Book';
+
+const Booklist = () => (
+  <div>
+    {books.map((book) => (
+      <div className="book-list" key={book.id}>
+        <h3>{book.title}</h3>
+        <p>{book.author}</p>
+        <p>{book.category}</p>
+        <button type="button" id={book.id}>delete</button>
+      </div>
+    ))}
```

Comment:
- [ ] You did a good job working on the previous reviewers' advice! 💯 However, I think you misunderstood the requirement:

As you can see, the jsx you have inside the map represents the UI for a single `Book`. So a good practice here is that you can extract a new component called `Book` and send the title, author, id, and category as props, something like this:

```js
const Book = ({id, title, author, category}) => {
  return (
    <div className="book-list">
      <h3>{title}</h3>
      <p>{author}</p>
      <p>{category}</p>
      <button type="button" id={id}>delete</button>
    </div>
  )
}
```

And then, you can refactor the code in your `BookList` component to this:

```suggestion
    {books.map((book) => (
      <Book key={book.id} id={book.id} title={book.title} author={book.author} category={book.category} />
      
      
      
      
      
    ))}
```

Ending up with a more readable and maintainable code! 🤓 🔧

Reply:
Thank you for the suggested changes, I have implemented the changes by extracting a new component called Book and sent the title, author, id, and category as props. I also validated the props.

Thank you once again